packages/design-system-tokens/lib/types.ts

```diff
@@ -0,0 +1,37 @@
+import { ImportCall } from 'typescript';
+
+export type HexValue = `#${string}`;
+export type RGBValue = `rgb(${number},${number},${number})`;
+export type PxValue = `${number}px`;
+
+export type SpacingToken = {
+  readonly [key: `spacer-${string}`]: PxValue;
+};
+
+export type HexColorToken = {
+  readonly [key: string]: HexValue;
+};
+
+export type Theme = {
+  readonly name: string;
+  readonly tokens: ThemeTokens;
+};
+
+export type ThemeTokens = {
+  readonly color: HexColorToken;
+  readonly spacing: SpacingToken;
+};
+
+export type FileDescriptor = {
+  //   m: moduleImportName,
+  m: string,
+  //   pd: parentDirectoryName,
+  pd: string,
+  //   bn: fileBaseName,
+  bn: string,
+  //   efn: exportFileName
+  efn: string,
+};
```

Comment:
If we have to explain what these properties mean, we should just use the long form. I generally try to avoid single-letter var names unless it's something like an incrementor. Clarity over brevity.

Reply:
I also think this type definition could move to the module that has the function that generates these objects.

Reply:
Agreed, updated, regarding it's location, it's used in several places and will be used in several places more, I don't think it makes sense to put it in the utilities.ts lib because of that.